datarequest/datarequest.py

```diff
@@ -3,15 +3,422 @@
+        return "Datamanager reject"
+    elif request_status == "DATAMANAGER_RESUBMIT":
+        return "Datamanager reject (resubmit)"
+    elif request_status == "UNDER_REVIEW":
+        return "Under review"
+    elif request_status == "REJECTED_AFTER_DATAMANAGER_REVIEW":
+        return "Rejected after datamanager review"
+    elif request_status == "RESUBMIT_AFTER_DATAMANAGER_REVIEW":
+        return "Rejected (resubmit) after datamanager review"
+    elif request_status == "REVIEWED":
+        return "Reviewed"
+    elif request_status == "APPROVED":
+        return "Approved"
+    elif request_status == "REJECTED":
+        return "Rejected"
+    elif request_status == "RESUBMIT":
+        return "Rejected (resubmit)"
+    elif request_status == "RESUBMITTED":
+        return "Resubmitted"
+    elif request_status == "DAO_APPROVED":
+        return "Approved (data assessment)"
+    elif request_status == "PREREGISTRATION_SUBMITTED":
+        return "Preregistration submitted"
+    elif request_status == "PREREGISTRATION_CONFIRMED":
+        return "Preregistration confirmed"
+    elif request_status == "DTA_READY":
+        return "DTA ready"
+    elif request_status == "DTA_SIGNED":
+        return "DTA signed"
+    elif request_status == "DATA_READY":
+        return "Data ready"
+
+#def build_object_path(path, relative_path, filename):
+#    relative_path = os.path.dirname(relative_path)
+#    path = path.lstrip("/")
+#
+#    # Build relative path.
+#    if relative_path:
+#    else:
+#        base_dir = os.path.join("/" + g.irods.zone, 'home', path)
+#
+#    return os.path.join(base_dir, filename)
+
+
+# Controllers
+
+@datarequest_bp.route('/archive')
+def index_archived():
+    return index(archived = True)
+@datarequest_bp.route('/dacrequests')
+def index_dacrequests():
+    return index(dacrequests = True)
+@datarequest_bp.route('/index')
+@datarequest_bp.route('/')
+def index(archived = False, dacrequests = False):
+    # Todo: read browser-items-per-page from config
+    items = 10
+
+    roles              = api.call('datarequest_roles_get')['data']
+    submission_allowed = 'PM' not in roles and 'DM' not in roles
+    is_dac_member      = 'DAC' in roles
+
+    return render_template('datarequest/index.html',
+                           activeModule='datarequest',
+                           items=items,
+                           archived=archived,
+                           dacrequests=dacrequests,
+                           submission_allowed=submission_allowed,
+                           is_dac_member=is_dac_member,
+                           help_contact_name='Todo_help_contact_name',
+                           help_contact_email='Todo_help_contact_email')
+
+
+@datarequest_bp.route('view/<request_id>')
+def view(request_id):
+    roles = api.call('datarequest_roles_get', {'request_id': request_id})['data']
+
+    if not 'PM' in roles and not 'DM' in roles and not 'DAC' in roles and not 'OWN' in roles:
```

Comment:
If we define this condition after the is_* declarations, we have a more readable conditon.